app/services/map_data_service.py

```diff
@@ -0,0 +1,561 @@
+        df = pd.read_csv(attraction_csv_path)
+        citys = df["City"]
+        attractions = df["Attraction"]
+        place_ids = []
+        for city, attraction in zip(citys, attractions):
+            place_id = self.get_place_id(f"{attraction}, {city}")
+            if place_id:
+                place_ids.append(place_id)
+        for place_id in place_ids:
+            processed_data = self.process_place_data(place_id)
+            self.save_processed_data(processed_data, f"{config.get('output_json_path')}/{place_id}.json")
+
+    def process_all_places_csv(self, attraction_csv_path: str) -> None:
+        try:
+            df = pd.read_csv(attraction_csv_path)
+            
+            if "City" not in df.columns or "Attraction" not in df.columns:
+                raise ValueError("CSV must contain 'City' and 'Attraction' columns")
+            
+            cities = df["City"]
+            attractions = df["Attraction"]
+            
+            all_processed_data = []
+            
+            for city, attraction in zip(cities, attractions):
+                try:
+                    place_id = self.get_place_id(f"{attraction}, {city}")
+                    if not place_id:
+                        print(f"⚠️  Could not find place_id for: {attraction}, {city}")
+                        continue
+                    
+                    processed_data = self.process_place_data(place_id)
+                    
+                    processed_data["name"] = attraction
+                    processed_data["city"] = city
+                    
+                    all_processed_data.append(processed_data)
+                    
+                except Exception as e:
+                    print(f"❌ Error processing {attraction}, {city}: {e}")
+                    continue
+            
+            if not all_processed_data:
+                return
+            
+            result_df = pd.DataFrame(all_processed_data)
+            
+            columns = ["name", "city", "place_id", "address", "phone_number", "url", 
+                      "coordinates", "rating", "reviews", "price_level", "tags", 
+                      "wheelchair_accessible_entrance", "opening_hours",
+                      "postal_code", "administrative_area_level_1", 
+                      "administrative_area_level_2", "sublocality_level_1",
+                      "serves_beer", "serves_breakfast", "serves_brunch", 
+                      "serves_dinner", "serves_lunch", "serves_vegetarian_food", 
+                      "serves_wine", "takeout"]
+            
+            existing_columns = [col for col in columns if col in result_df.columns]
+            result_df = result_df[existing_columns]
+            
+            output_path = config.get("output_csv_path")
+            result_df.to_csv(output_path, index=False, encoding='utf-8')
+            
+        except Exception as e:
+            raise ValueError(f"Error processing CSV: {e}")
```

Comment:
1. Check for file existence (using `os.path.exists`) before opening files to avoid uncaught exceptions.
2. For `process_all_places_csv`, validate all required columns before looping.
3. *Optional* Consider using `tqdm` or similar for progress indication if running large batches (for CLI tools). https://github.com/tqdm/tqdm